### PR TITLE
Add tests for std.Thread.Condition and std.Thread.Semaphore

### LINKED ISSUE #10636: add basic test coverage for std.Thread.Condition and std.Thread.Semaphore
There is no test coverage for these yet!

### PR DESCRIPTION
**Pull request by naeu**
Fixes #10636

### CODE REVIEW

File: lib/std/Thread/Condition.zig [hunk 1]
```diff
@@ -193,3 +194,47 @@ pub const AtomicCondition = struct {
             waiter.data.notify();
     }
 };
+
+test "basic test" {
+    if (builtin.single_threaded) {
+        return;
+    }
```
Comment (ominitay) [inline lib/std/Thread/Condition.zig:201]:
```suggestion
test "Thread.Condition" {
    if (builtin.single_threaded) {
        return error.ZigSkipTest;
    }
```

File: lib/std/Thread/Semaphore.zig [hunk 2]
```diff
@@ -31,3 +33,29 @@ pub fn post(sem: *Semaphore) void {
     sem.permits += 1;
     sem.cond.signal();
 }
+
+test "basic test" {
+    if (builtin.single_threaded) {
+        return;
+    }
```
Comment (ominitay) [inline lib/std/Thread/Semaphore.zig:40]:
```suggestion
test "Thread.Semaphore" {
    if (builtin.single_threaded) {
        return error.ZigSkipTest;
    }
```

**Review by ominitay — CHANGES_REQUESTED** [review]

**Comment by ominitay** [comment]
Apologies, should have been `SkipZigTest`, I shuffle words a lot

**Comment by naeu** [comment]
No worries, updated.

**Comment by ominitay** [comment]
Congrats on your first contribution :)

**Comment by naeu** [comment]
In case it's worth fixing, [there's a branch](https://github.com/naeu/zig/commits/thread-test-fix) to make these tests more robust and avoid spurious wakes in the condition test.

**Pull request merged**